workflow/src/main/kotlin/com/r3/corda/sdk/token/workflow/flows/AddUpdateEvolvableToken.kt

```diff
@@ -0,0 +1,37 @@
+package com.r3.corda.sdk.token.workflow.flows
+
+import co.paralleluniverse.fibers.Suspendable
+import com.r3.corda.sdk.token.contracts.commands.Update
+import com.r3.corda.sdk.token.contracts.states.EvolvableTokenType
+import net.corda.core.contracts.StateAndRef
+import net.corda.core.flows.FlowLogic
+import net.corda.core.identity.Party
+import net.corda.core.transactions.TransactionBuilder
+
+/**
+ * Helper flow for assembling an [Update] transaction for [EvolvableToken]s. This accepts a [TransactionBuilder] as well
+ * as old and new states, and adds commands, inputs, and outputs to the transaction. Returns the [TransactionBuilder]
+ * to allow for continued assembly of the transaction, as needed.
+ */
+class AddUpdateEvolvableToken(
```

Comment:
This could actually be a function :) We were having a discussion the other day about when flows should be used vs just regular functions. I guess we only need to use flows if a sub flow needs calling or a message needs to be sent or received from another node. I like where this is going though - i feel it would be powerful to make most of the flows/functions things which mutate and return a tx builder.

Reply:
Yes, I saw those comments and opted to implement here. Flow or function, can go either way! Opted to make it a flow for maximum extensibility and capability - even if it is overkill for this particular instance.

Reply:
Hi Roger, please let me know if we have a preference for this being a function or a flow?

Reply:
function works best (if no flow logic methods are required)